Addressed comments on s8 task observed during testing

rsarwad opened the pull request:
"[lte][agw]: Addressed comments on s8 task observed during testing

## Summary
1. Added ULI and serving network in deletion session request
2. Added pco in create session request and processing pco which is received in create session response
3. Removed explicit length and num_protocol_container_id field in pco
4. Fixed the memory leak found during testing
## Test Plan
Executed s1ap sanity test suite

uri200 committed:
[feg] s8_proxy add ULI to Delete Request

Signed-off-by: Oriol Batalla <obatalla@fb.com>

uri200 committed:
[feg] s8_proxy change c_pgw_fteid (fteid) to c_pgw_teid (teid)

Signed-off-by: Oriol Batalla <obatalla@fb.com>

rsarwad committed:
Merge remote-tracking branch 'upstream/master' into rsarwad_address_comments_on_s8

Signed-off-by: rashmi <rashmi.sarwad@radisys.com>

uri200 commented on lte/gateway/c/oai/lib/s8_proxy/s8_client_api.cpp:
```diff
@@ -455,12 +501,17 @@ static void convert_proto_msg_to_itti_csr(
+  if (response.has_protocol_configuration_options()) {
+    get_pco_from_proto_msg(
+        response.protocol_configuration_options(), &s5_response->pco);
+  }
```
I can't comment the line 499, but shoudn't we be checking `if (response.has_gtp_error())` as the very beginning of this function? 

If we have gtp_error, the other field are going to be bad. Only gtp_error is good. So no need to go through the whole function.

uri200 commented on lte/gateway/c/oai/tasks/mme_app/mme_app_itti_messaging.c:
```diff
@@ -361,19 +361,8 @@ int mme_app_send_s11_create_session_req(
-  session_request_p->serving_network.mnc[2] =
-      ue_mm_context->e_utran_cgi.plmn.mnc_digit3;
+  COPY_PLMN_IN_ARRAY_FMT(
+      (session_request_p->serving_network), (ue_mm_context->e_utran_cgi.plmn));
```
👍

rsarwad committed:
Merge remote-tracking branch 'upstream/master' into rsarwad_address_comments_on_s8

Signed-off-by: rashmi <rashmi.sarwad@radisys.com>

rsarwad committed:
1. Added code to send ULI and serving network in delete session request
2. Added code to send and receive PCO in create session req/response
3. Fixed memory leaks found during testing

Signed-off-by: rashmi <rashmi.sarwad@radisys.com>

ulaskozat commented on lte/gateway/c/oai/lib/s8_proxy/s8_client_api.cpp:
```diff
@@ -195,6 +195,30 @@ static void recv_s8_create_session_response(
+    mnc[2]  = convert_digit_to_char(itti_msg_sn.mnc[2]);
+    mnc_len = 3;
+  } else {
+    mnc[2]  = '\0';
```
looks already initialized to null (ascii 0) on line 202.

ulaskozat reviewed (approved):
LGTM
